TriblerGUI/widgets/trustpage/js/drawing.js

```diff
@@ -133,24 +191,35 @@ function drawLinks(svg, data) {
         .attr("class", "link")
         .style("opacity", "0");
 
+    // A link between two nodes is composed of two parts, representing the amount of data uploaded and downloaded
+
+    // The source part of the link
     links.append("line")
         .attr("class", "link-source")
-        .attr("stroke-width", function (d) {
-            return getStrokeWidth(d, data)
-        })
         .style("stroke", config.link.colorLinkSource);
 
+    // The target part of the link
     links.append("line")
         .attr("class", "link-target")
-        .attr("stroke-width", function (d) {
-            return getStrokeWidth(d, data)
-        })
         .style("stroke", config.link.colorLinkTarget);
 
-    links.append("line")
-        .attr("class", "link-velocity")
-        .attr("stroke-width", 2)
-        .style("stroke", "white");
+    // Functionality for the composed link
+    d3.selectAll(".link")
```

Comment:
Please change this to ```links```.

Now, everytime you call ```drawLinks```, it adds .on(mouseover) etc. to all links, instead of just the new ones. The old ones now have two exactly the same event listeners (the callback is executed twice). This keeps increasing when you click new nodes.